app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/RedirectHelper.java

```
@@ -21,6 +21,7 @@
 public class RedirectHelper {
 
     public static final String DEFAULT_REDIRECT_URL = "/applications";
+    public static final String DEFAULT_REDIRECT_ORIGIN = "https://app.appsmith.com";
```

How would this work in on-prem scenarios?  Could we instead error out  in case redirect origin is not present?

---

So the problem here is that once we enter the flow, the client does not stick around for an error response either. Which means any error that is caught by us will need to be redirected back to a known url. If I add the error scenario, we'd end up with a plain json body on the page. Would that work? @trishaanand

---

I can see that you are deriving the redirect domain based on the `Origin` header in the function `getRedirectDomain`. Is there any scenario where you think this fail-safe will kick into action? If not, I think using app.appsmith.com as a fail-safe is OK.

---

For the current use case we'd be expecting the `Referer` header (GET request that is not CORS). I've added the other options for fail-safes on that. This constant is fail-safe layer 2 😆  

So no, not aware of when we'd hit it.